### DISCUSSION TITLE
Normalization/Rarefaction for Distance Matrix-based Metrics

### DISCUSSION THREAD

**Discussion by todd-testerman** [david-barnett/microViz#31] [Q&A] [answered]
Hi David,

First off, wanted to say I've really enjoyed using your package thus far! The ability to jump between the Shiny app and R to customize figures has been amazing. I did have one question regarding ordinations using distance matrices.

- The recommended transformation (or lack thereof) in the tutorial information is "identity" when generating distance matrices (like when using Bray-Curtis). At this point, I have not accounted for differing library sizes. Traditionally, I am used to either rarefying or doing some other kind of transformation to account for these differences before running a distance metric. I realize some of these transformations present their own issues but I am basically wanting to clarify whether I should be doing this before running a distance metric in microViz? If not, would there be a good resource to reference regarding using non-normalized read counts in metrics such as these?

Thanks again for the help!

TT

**Comment by david-barnett** [comment, marked as answer]
Hi Todd,

Glad you have liked it so far. 

This is a tricky question to answer definitively, but there has been some discussion about rarefying vs. not. Here are a couple of relevant articles.

[Normalization and microbial differential abundance strategies depend upon data characteristics.](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0237-y) Weiss 2017 
[Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible](https://doi.org/10.1371/journal.pcbi.1003531). McMurdie 2014  

I think how important these decisions are will depend on the dissimilarity measure you would want to use (as well as how substantial the difference in read depth is across your samples). 
For dissimilarities that rely on the unweighted presence or absence of taxa like unweighted unifrac or binary jaccard, doing anything that affects/removes low abundance taxa is going to make a bigger difference to your ordination results than for say, weighted unifrac, or bray-curtis, or aitchison, for which differences in low abundance taxa matter less. To me, this ends up being a practical reason to avoid unweighted measures.

So you can try out prevalence/abundance filtering at different levels with `tax_filter`, and phyloseq has a rarefying procedure too you can use easily in a microViz workflow. https://rdrr.io/bioc/phyloseq/man/rarefy_even_depth.html Then you can see how much these choices will matter for your dataset and ordination method.

Because you mentioned Bray-Curtis, I also remembered this specific little blog post that I found interesting: 
[Bray-Curtis dissimilarity on relative abundance data is the Manhattan distance (aka L₁ distance)](https://metarabbit.wordpress.com/2018/01/16/bray-curtis-dissimilarity-on-relative-abundance-data-is-the-manhattan-distance-aka-l%E2%82%81-distance/)

Hope some of these thoughts might be helpful to you. 

cheers,
David
